### PR TITLE
[ENH] metrics rework part I - output format tests

### PR DESCRIPTION
**Pull request by fkiraly**
This PR is part I of a rework of metrics.

It introduces output format tests to enable testing in the refactoring process:
* the current `numpy` based functions in the `performance_metrics._functions` module
* the classes in the `performance_metrics._classes` module

This is necessary since refactoring should be test anchored, and the current tests for metrics did not test the output format.

Contains fix of a bug discovered by this:
* `geometric_mean_squared_error` and `geometric_mean_absolute_error` were inconsistent with the general intervace, missing `kwargs`. This was added.

### CODE REVIEW

File: sktime/performance_metrics/tests/test_metrics_classes.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    res = dict()
+
+    res[1] = metric(multioutput=multioutput)(
+        y_true=y_true,
+        y_pred=y_pred,
+        y_pred_benchmark=y_pred,
+        y_train=y_true,
+    )
+
+    res[2] = metric(multioutput=multioutput)(
```
Comment (eenticott-shell) [inline sktime/performance_metrics/tests/test_metrics_classes.py:40]: This is the same as above, is it meant to be `metric.evalute()`?

File: sktime/performance_metrics/tests/test_metrics_classes.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+metric_classes = getmembers(_classes, isclass)
+
+exclude_starts_with = ("_", "Base")
+metric_classes = [x for x in metric_classes if not x[0].startswith(exclude_starts_with)]
+
+names, metrics = zip(*metric_classes)
+
+y_pred = _make_series(n_columns=2, n_timepoints=20, random_state=21)
+y_true = _make_series(n_columns=2, n_timepoints=20, random_state=42)
```
Comment (eenticott-shell) [inline sktime/performance_metrics/tests/test_metrics_classes.py:19]: Is it worth creating two tests sets, one univariate and one multivariate?

Comment (eenticott-shell) [inline sktime/performance_metrics/tests/test_metrics_classes.py:19]: Is it worth creating two tests sets, one univariate and one multivariate?

Comment (fkiraly) [reply sktime/performance_metrics/tests/test_metrics_classes.py:19]: yes, good idea

Comment (fkiraly) [reply sktime/performance_metrics/tests/test_metrics_classes.py]: added them

**Review by ltsaprounis — APPROVED** [review]

**Pull request merged**